### PR TITLE
Add the NT-3

### PR DESCRIPTION
**Pull request by SirWarock**
## About the PR

I added the NT-3, a new NT superweapon that has been picked up out of the Old War dumpsters, refurbished by cheapskate NT to skip on providing actual equipment to their security Staff.
I added a new magazine type, light rifle small box for said superweapon.
I added a Bipod System, where you can set up a bipod with a short doafter, after which you gain increased firerate and accuracy.
## Why / Balance

We're going to lose the Gamblagator and the Beam Devastator, so I thought, we should have some other super weapons.
Unlike the laser variants, this one fits right in with the RPG-7, also an old war weapon and fits the theme of cheapskate NT.

Functionally, it's worse than the Vulcan except for double the magazine size.
However, if you put the bipod down, it'll become very accurate and the firerate is increased, but you are locked in place.
It spawns with 3 boxes total, including the one spawning in the gun. So if you want to keep a supression fire going, you'll need to cumbersomly reload the magazine box.
## Technical details

Added the `GunBipodComponent.cs` and `SharedGunSystem.Bipod.cs` for the Bipod Mechanics.
Added various Sprites for the NT-3 and the Bipod Action.
Added the NT to the Superweapon Table.
Added the Action to setup your Bipod.
Added the small light rifle box.
## Media

grafik
grafik
grafik
grafik
grafik

https://github.com/user-attachments/assets/f2a03f95-fbe4-44f3-8297-1c8b2f82248e

https://github.com/user-attachments/assets/ce54678e-653b-454a-9131-a04152eae7ef

## Requirements

- [X] I have tested all added content and changes.
- [X] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- add: Added the NT-3, a brand new, top of the line, science-breakthrough weapon! ...or that's what NT wants you to think. Obtainable through Cargo for 13000 spesos.

### CODE REVIEW

**Comment by TehFlaminTaco** [comment]
Fails real

**Comment by SirWarock** [comment]
All test fails are assumed fake until proven real🔥

**Comment by einknusprigestoast** [comment]
Approved by Direction

**Comment by SirWarock** [comment]
No 24 hour direction?

**Comment by Cepelinas1** [comment]
sorry for the massive delay, you've managed to cause a big debate on what's a superweapon and what's not
give us a bit more time, we like the weapon's idea

**Comment by Vapetastic-Gaming** [comment]
Seems the consensus is:

Make this a purchasable Cargo weapon, It doesnt match well with our Superweapons, but should still be added

**Comment by kotobdev** [comment]
> Make this a purchasable Cargo weapon

gertrude is frothing at the mouth

jokes aside. this will be absolute antag heaven. 4tc money kit + 4tc emag for a free superweapon. lol

**Comment by SirWarock** [comment]
How much will it cost?
If it costs too much, I'm afraid this will be a traitor weapon only..
I coded this to have fun with it, not solely be screwed over by it as a Security player.

**Comment by Woljif** [comment]
> How much will it cost? If it costs too much, I'm afraid this will be a traitor weapon only.. I coded this to have fun with it, not solely be screwed over by it as a Security player.

I'm of the opinion that it should be a superweapon. With the removal of the beam dev and Gatorgun, it seems like we're moving towards less swingy options that are strong but not trump cards. Everyone except sec players kind of hates superweapons, and I say this as someone who's been on both sides of every one.

I think this is the perfect level of power for a superweapon to have that's not uber-swingy. It's essentially an L6 saw that you have to remain in place for. It's not an all-or-nothing RPG shot, it's not a 120 damage hitscan shot that goes through windows from 30 tiles off your screen, etc.

**Comment by Woljif** [comment]
> > Make this a purchasable Cargo weapon
> 
> gertrude is frothing at the mouth
> 
> jokes aside. this will be absolute antag heaven. 4tc money kit + 4tc emag for a free superweapon. lol

Gertrude playing cargo assistant specifically for this (spend 17tc on a c20 or 4tc on a budget L6?)

**Comment by SirWarock** [comment]
I personally think it matches well with the super weapon we have (The RPG). Both are old ancient weaponry.
I'd go as far and say that the Gamblagator and the Beam Devastator do not match the design of the original superweapon, the RPG.

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+using Content.Shared.DoAfter;
+using Content.Shared.Hands;
+using Content.Shared.Hands.Components;
+using Content.Shared.IdentityManagement;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Maps;
+using Content.Shared.Toggleable;
+using Content.Shared.Weapons.Ranged.Events;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Serialization;
```
Comment (Toby222) [inline Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: unused usings `Content.Shared.Maps` and `Robust.Shared.Physics.Systems`

```suggestion
using Content.Shared.Hands.Components;
using Content.Shared.IdentityManagement;
using Content.Shared.Interaction.Events;
using Content.Shared.Toggleable;
using Content.Shared.Weapons.Ranged.Events;
using Robust.Shared.Serialization;
```

File: Resources/Prototypes/_DV/Actions/types.yml [hunk 2]
```diff
@@ -66,3 +66,13 @@
     useDelay: 30
   - type: InstantAction
     event: !type:CoughItemActionEvent
+
+- type: entity
+  parent: BaseToggleAction
+  id: ActionToggleBipod
+  name: Setup your bipod
```
Comment (Toby222) [inline Resources/Prototypes/_DV/Actions/types.yml]:
```suggestion
  name: Set up your bipod
```

File: Resources/Prototypes/_DV/Actions/types.yml [hunk 3]
```diff
@@ -66,3 +66,13 @@
...
+  description: Increase the accuracy and firerate of your firearm.
```
Comment (Toby222) [inline Resources/Prototypes/_DV/Actions/types.yml]:
```suggestion
  description: Increase the accuracy and fire rate of your firearm.
```

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+        var blockQuery = GetEntityQuery<GunBipodComponent>();
+        var handQuery = GetEntityQuery<HandsComponent>();
+
+        if (!handQuery.TryGetComponent(args.Performer, out var hands))
+            return;
+
+        var bipods = _hands.EnumerateHeld((args.Performer, hands)).ToArray();
+
+        foreach (var bipod in bipods)
```
Comment (Toby222) [inline Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: Redundant `ToArray` just iterates twice
```suggestion
        foreach (var bipod in _hands.EnumerateHeld((args.Performer, hands)))
```

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    private void OnToggleAction(Entity<GunBipodComponent> ent, ref ToggleActionEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        var blockQuery = GetEntityQuery<GunBipodComponent>();
+        var handQuery = GetEntityQuery<HandsComponent>();
+
+        if (!handQuery.TryGetComponent(args.Performer, out var hands))
+            return;
```
Comment (Toby222) [inline Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: No need to get blockQuery if the handQuery fails
```suggestion
        var handQuery = GetEntityQuery<HandsComponent>();

        if (!handQuery.TryGetComponent(args.Performer, out var hands))
            return;

        var blockQuery = GetEntityQuery<GunBipodComponent>();
```

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        if (xform.GridUid != xform.ParentUid)
+        {
+            CantSetupError(user);
+            return;
+        }
+
+        // Don't allow someone to set up the bipod if they're not holding the weapon
+        if (!_hands.IsHolding(user, ent.Owner, out _))
+        {
+            CantSetupError(user);
```
Comment (Toby222) [inline Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: Is it possible to give this a separate error message? "There was nothing to set up on" when the reason is that it's on your back is a bit annoying

File: Content.Shared/_DV/Weapons/Ranged/Components/GunBipodComponent.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class GunBipodComponent : Component
+{
+    [ViewVariables(VVAccess.ReadWrite), DataField("minAngle"), AutoNetworkedField]
+    public Angle MinAngle = Angle.FromDegrees(-43);
+
+    /// <summary>
+    /// Angle bonus applied upon the bipod being used.
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite), DataField("maxAngle"), AutoNetworkedField]
```
Comment (Toby222) [inline Content.Shared/_DV/Weapons/Ranged/Components/GunBipodComponent.cs]: Redundant attributes
```suggestion
    [DataField, AutoNetworkedField]
    public Angle MinAngle = Angle.FromDegrees(-43);

    /// <summary>
    /// Angle bonus applied upon the bipod being used.
    /// </summary>
    [DataField, AutoNetworkedField]
```

**Review by Toby222 — CHANGES_REQUESTED** [review]

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs (hunk 6 above)
Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: It's moreso an edge case, I think. You cannot activate the button if you do not hold it, I believe. Let me test that.

Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: Nope, you're entirely correct.

Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: Done!

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs (hunk 4 above)
Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: I removed this. It served to ensure that you can't use more than one bipod.
Why though? If someone adds more weapons with bipods later on, why not let them one-hand them?
So I instead rewrote the code to allow it.
(The NT-3 cannot be shot without being wielded, so you cannot have two NT-3s with bipods.)

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs (hunk 5 above)
Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: Rewrote the code to allow multiple Bipods to be used.

File: Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs (hunk 1 above)
Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Systems/SharedGunSystem.Bipod.cs]: You are a hawk. Removed them! 👍

File: Content.Shared/_DV/Weapons/Ranged/Components/GunBipodComponent.cs (hunk 7 above)
Comment (SirWarock) [reply Content.Shared/_DV/Weapons/Ranged/Components/GunBipodComponent.cs]: So true, Magister

**Comment by SirWarock** [comment]
Changelog:
- Emplacing the Bipod does not anchor you anymore.
This allows you to move, as well as be moved by exterior forces (Spacewind, someone dragging you, etc.) Previously, you were simply immovable.
- Moving automatically packs up the bipod.
This is for convenience sake. Having to press an action button to undo the bipod to move when you wanna move is bad design.
But this also allows someone with, for example the homerun bat, to move you and undo your bipod.
- It no longer blocks using multiple bipods.
If someone else wants to add bipods to guns, they might make a one-handed bipod. Nothing says you can't have two bipods usable. So I removed the check that would otherwise block it.
**Noticably, the NT-3 does need to be wielded to be shot. You cannot have two NT-3s with active Bipods.**

**Comment by Vova567810** [comment]
Insert rifleman with heavy m54c yard stare

**Comment by Cepelinas1** [comment]
okay jesus christ, this took forever to agree on
generally direction likes it, but we have changes if you can make them:

- make this a cargo purchasable weapon, for 13k. Have it the weapon and like a spare mag available. This is kinda chosen, due to us not wanting the funny situation for security, where they have 3 NT-3s and 1 drozd.
- make it use a new caliber, .40. The box for refills (not magazines) can be printed by security by like 10 steel and some uranium, but only reloads 25 rounds. These rounds should generally do like 20 pierce and 20 structural
- when the bipod is deployed, NT-3 should have a firerate of 7, when not: 4
- **now for the hard part**: when fired, these rounds (or just NT-3) have the ability to pierce through walls/doors/windows, but not reinforced walls, shuttle walls or anything reinforced kind.

the whole debate took so long, because we were arguing for what is a good "superweapon". We originally liked the premise, but disliked that this was another superweapon that didn't have any real danger to it's surroundings or any reason to **use it only in an emergency** and not just for a ninja. This conversation went back and forth, before we agreed on the stats, which then went into "where should it appear". We talked about it being a seperate safe spawn along security, but then it would be another good gun in armory and be weird thematically. If it was another rng safe weapon spawn, it would suck ass if it's nukeops, but you only have 4 mgs in the safe and practically impossible to push the attackers.

generally, we want to see this test-merged and just improve it along the way, tweaking or adding/removing things about it. We like the idea, we just don't know how to implement it without being too shit or too op

**Comment by Cepelinas1** [comment]
feel free to message me on discord if you wanna talk about it and stuff

**Comment by dekigangsta** [comment]
> okay jesus christ, this took forever to agree on generally direction likes it, but we have changes if you can make them:
> 
>     * make this a cargo purchasable weapon, for 13k. Have it the weapon and like a spare mag available. This is kinda chosen, due to us not wanting the funny situation for security, where they have 3 NT-3s and 1 drozd.
> 
>     * make it use a new caliber, .40. The box for refills (not magazines) can be printed by security by like 10 steel and some uranium, but only reloads 25 rounds. These rounds should generally do like 20 pierce and 20 structural
> 
>     * when the bipod is deployed, NT-3 should have a firerate of 7, when not: 4
> 
>     * **now for the hard part**: when fired, these rounds (or just NT-3) have the ability to pierce through walls/doors/windows, but not reinforced walls, shuttle walls or anything reinforced kind.
> 
> 
> the whole debate took so long, because we were arguing for what is a good "superweapon". We originally liked the premise, but disliked that this was another superweapon that didn't have any real danger to it's surroundings or any reason to **use it only in an emergency** and not just for a ninja. This conversation went back and forth, before we agreed on the stats, which then went into "where should it appear". We talked about it being a seperate safe spawn along security, but then it would be another good gun in armory and be weird thematically. If it was another rng safe weapon spawn, it would suck ass if it's nukeops, but you only have 4 mgs in the safe and practically impossible to push the attackers.
> 
> generally, we want to see this test-merged and just improve it along the way, tweaking or adding/removing things about it. We like the idea, we just don't know how to implement it without being too shit or too op

"make this a cargo purchasable weapon"
im-out-1-seconds-shark-tank-shark-tank

**Comment by SirWarock** [comment]
## Changelog:
- The fire rate has been reduced to 4 without bipod and 7 with bipod.
- It's now a cargo weapon, available for purchase for 13k Spesos.

grafik

It comes with the following:

grafik

- The ammunition used by the NT-3 is .40 Anti-Materiel, stored in 60 capacity magazines and 30 capacity ammo boxes, which can be printed for 10 steel and 2.5 uranium.

grafik
grafik

- The new .40 Anti-Materiel deals 20 Piercing & Structural, as well as pierces Structures and anything that isn't one.
Currently, it pierces one wall/window. It also accounts for when they are a row - Walls of the same row are ignored.
In this example, the blue path is valid, crossing two walls and piercing both - The red path is invalid, the second wall is considered a new row of walls and will block the bullet.

grafik
grafik

This does not pierce structures with a health over 200 (Normal walls have 200, reinforced have 400.) However, if you keep shooting long enough, you'll reduce the health of a reinforced wall enough to be able to pierce it eventually, before breaking it entirely in the end.

https://github.com/user-attachments/assets/98e3522b-45e9-4209-8a13-60b58b6e4ee2

**Comment by SirWarock** [comment]
I feel so sorry for the Maintainers - Whoop, pinged someone I didn't mean to ping

**Comment by Toby222** [comment]
> I feel so sorry for the Maintainers - Whoop, pinged someone I didn't mean to ping

-> Adds one item
-> 46 files changed
:Godo:

**Comment by SirWarock** [comment]
> -> Adds one item
>-> 46 files changed :Godo:

You should probably just let Direction take a look at it before you look over it - In case they request more changes

**Comment by Cepelinas1** [comment]
alright, direction approved for a test merge
sorry for taking it so long

if it isn't used that often, we might change where to get it

**Review by ShepardToTheStars — APPROVED** [review]
Very well coded.

The one (very small) thing I would probably change is to take the bipod icons out of the NT-3 RSI directory and put in its own but that can be done if another gun ever gets the bipod component in the future, or if you are so inclined to clean that up.

**Comment by ShepardToTheStars** [comment]
@Toby222 Requires your approval before it can be merged (or at least to clear your requesting changes)

**Review by Toby222 — APPROVED** [review]
